### PR TITLE
- Low power mode via trigger

### PR DESCRIPTION
**Pull request by GANDAMU**
Copy/pasted from code comments:
```
    To go into low power mode:

        oscsend localhost 5006 /trigger/low_power_mode i 1

    To go into normal power mode:

        oscsend localhost 5006 /trigger/low_power_mode i 0
```

Backup gets saved in same directory that rosary.py resides in, e.g. `MegaPrayer-oscled/python/mp/low_power_mode.json`

### CODE REVIEW

File: python/mp/rosary.py [hunk 1]
```diff
@@ -53,6 +56,72 @@ def update(self):
...
+        msg.add_arg(int(0))                     # what these two do
+        msg.add_arg(float(3.0))                 # But I get this one!
+
+        for i in range(3):
+            if low_power_mode:
+                msg.add_arg(1.0)
+            else:
+                msg.add_arg(.75)
+
+        self.osc_client.send(msg.build())
```
Comment (GANDAMU) [inline python/mp/rosary.py:114]: @jdblair You're probably gonna wanna check this part.

File: python/mp/rosary.py [hunk 2]
```diff
@@ -53,6 +56,72 @@ def update(self):
...
+
+        self.write_to_disk()
+
```
Comment (jdblair) [inline python/mp/rosary.py]: Please move this into the trigger handler so its only written once per state change. I don't want to wear out the flash memory faster than necessary.

File: python/mp/triggers/low_power_mode.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    """
+
+    # Need to give it a default so it doesn't die on registration
+    def __init__(self, enable_low_power_mode=False):
+        super().__init__("low_power_mode")
+        # Turn the int/float that we get into a bool
+        self.low_power_mode = enable_low_power_mode > 0
+
+    def inner_fire(self):
+        self.rosary.low_power_mode = self.low_power_mode
```
Comment (jdblair) [inline python/mp/triggers/low_power_mode.py]: Can we call `self.rosary.write_to_disk()` here?

**Review by jdblair — CHANGES_REQUESTED** [review]
Looks good, except for writing the state every 10 seconds.

File: python/mp/triggers/low_power_mode.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    # Need to give it a default so it doesn't die on registration
+    def __init__(self, enable_low_power_mode=False):
+        super().__init__("low_power_mode")
+        # Turn the int/float that we get into a bool
+        self.low_power_mode = enable_low_power_mode > 0
+
+    def inner_fire(self):
+        # Directly update the PowerModeUpdater object
+        self.rosary.power_mode_updater.low_power_mode = self.low_power_mode
+        self.rosary.power_mode_updater.write_to_disk()
```
Comment (GANDAMU) [inline python/mp/triggers/low_power_mode.py:27]: Alright, writing from here now.

File: python/mp/rosary.py [hunk 5]
```diff
@@ -236,6 +302,12 @@ def beads_set_bgcolor(self, beads):
     def osc_server_main(self):
         print("Serving on {}".format(self.osc_server.server_address))
         self.osc_server.serve_forever()
+
+    def power_mode_loop(self):
+        print("Starting loop to update + backup power mode state")
+        while True:
+            self.power_mode_updater.update()
```
Comment (GANDAMU) [inline python/mp/rosary.py:306]: Just updating using whatever power_mode_updater's own low_power_mode value is.

**Review by GANDAMU — COMMENTED** [review]
Yeah, I really was just doing it every 10 seconds to be safe. Did play with the idea of only writing on triggers, but figured it wouldn't hurt to just write a bit more. Turns out, it would.

Tested locally by just firing the trigger and checking the file. Still not sure if the OSC message is correct, but if you're good I'm good.

File: python/mp/triggers/low_power_mode.py (hunk 4 above)
Comment (jdblair) [reply python/mp/triggers/low_power_mode.py:27]: thanks

**Comment by GANDAMU** [comment]
Okay, @jdblair I *think* that's what you mean when you say "remove them." If I'm misinterpreting, you can just go ahead and fix that. I'm not like super possessive about this branch or anything.

**Pull request merged**